src/System.IO.Compression/src/System/IO/Compression/GZipStream.cs

```diff
@@ -150,6 +150,18 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        public override ValueTask DisposeAsync()
+        {
+            if (GetType() != typeof(GZipStream))
+            {
+                return base.DisposeAsync();
+            }
+
+            DeflateStream ds = _deflateStream;
+            _deflateStream = null;
+            return ds.DisposeAsync();
```

Comment:
Potential NRE if DisposeAsync is called multiple times?

Reply:
Thanks.  Will fix.